data_processor.py

```diff
@@ -0,0 +1,101 @@
+# title           : data_processor.py
+# description     : AWS Lambda script;
+#                   processes manual uploads to manual ingest bucket
+# author          : Volpe Center (https://www.volpe.dot.gov/)
+# license         : MIT license
+# ==============================================================================
+#
+
+from __future__ import print_function
+
+import logging
+import boto3
+import urllib
+import os
+import time
+import json
+from datetime import datetime
+
+CHUNK_SIZE = 1073741824 # 1 GB
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def lambda_handler(event, context):
+    source_bucket = event['Records'][0]['s3']['bucket']['name']
+    raw_source_key = event['Records'][0]['s3']['object']['key']
+    logging.info(f'raw_source_key: {raw_source_key}')
+    source_key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'])
+    logging.info(f'source_bucket: {source_bucket}')
+    logging.info(f'source_key: {source_key}')
+
+    file_size = event['Records'][0]['s3']['object']['size']
+
+    target_bucket = os.environ['TARGET_DATA_BUCKET']
+    target_data_folder_path = bucket_destination_mapping()[source_bucket]

```

Comment:
?? Would it make sense to put the raw-submissions bucket within the BUCKET_PATH_MAPPING dictionary as well as the folder_path?  I know right now we have a single raw-submissions bucket, but if down we need to split the data-lake up into one bucket per data provider for example....  Just a thought

Reply:
Yes thats an excellent point, one I'm hoping to resolve with Dan in the meeting I have in one minute.

If we do end up splitting out buckets then I can look into streamlining this.